### PR TITLE
Closes #1103: Add new AZ News view displays

### LINKED ISSUE #1103: Add News view to match C/A "Content Hub" mockups
## Motivation
Casual Astronaut frequently includes a "Content Hub" section in their mockups. This commonly includes a category filter, custom view title, featured article, and category tags on the news teasers (separate issue).

### Screenshot
image

## Solution
- [ ] Add a new view to News that includes a category filter for existing taxonomy
- [ ] #1500
- [ ] Add new view to News using the Feature display mode
- [ ] Connect the view displays into one /content-hub page

### PR DESCRIPTION
**Pull request by danahertzberg**
## Description

Adds new view displays to AZ News view
- Feature: block with 1 news article sorted by sticky first, then publication date
- Grid with filter: block with 9 news articles sorted by publication date, filterable by category

## Related issues

Closes #1103 
**BLOCKED BY #1500** 

## How to test

1. Create a news article with older publication date
2. Make sticky at top of lists (right column)
3. Create a new page
4. Add view > AZ News > Feature
5. Verify sticky news article appears
6. Add view to page > AZ News > Grid with Filter
7. Verify filter works
8. Change items per page in view settings on page
9. Verify items per page works
10. Verify pager works
11. Add other AZ News views 
12. Verify no inadvertent changes to other AZ News views

### Note:
_These are **block** displays. No page at a searchable path will be made. These views are not attached/connected to each other._ 
#### Reasoning
- Adding another page to the site in addition to /news and /news-teasers starts to create bloat and confusion for end-users
- Two views can be added to a page and better managed by content editors
- Separate blocks provides more flexibility - can use one or the other or both

### Arizona Quickstart (install profile, custom modules, custom theme)
- Patch release changes
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
- Minor release changes
   - [x] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- [ ] Other or unknown

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation. (QS site)
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by trackleft** [comment]
```                 
189  [Drupal\Core\Config\UnmetDependenciesException]                              
190  Configuration objects provided by <em class="placeholder">az_news</em> have  
191   unmet dependencies: <em class="placeholder">views.view.az_news (core.entit  
192  y_view_mode.node.az_feature, better_exposed_filters)</em>                    
```

**Comment by trackleft** [comment]
Add better exposed filters as a dependency in the info file.

**Comment by trackleft** [comment]
https://00af7bc9-e13e-4e0c-9ea3-b6469900c4da.probo.build/featured-news 
I selected the feature view for az news, but see media lists

**Comment by trackleft** [comment]
I don't think I see the feature view mode https://7b5e2985-cedf-4a13-ab91-3a35916c835d--pr-1539.probo.build/test-chris

**Comment by danahertzberg** [comment]
I figured out the issue @trackleft. New commit made, waiting for probo to build. Thank you!

**Comment by danahertzberg** [comment]
Fixed the media list issue, have a php error, and styling now isn't rendering as expected.

File: modules/custom/az_news/config/install/views.view.az_news.yml [hunk 1]
```diff
@@ -84,7 +87,7 @@ display:
         type: 'entity:node'
         options:
           relationship: none
-          view_mode: az_media_list
+          view_mode: az_feature
```
Comment (trackleft) [inline modules/custom/az_news/config/install/views.view.az_news.yml:90]: Are we sure we want this to update existing sites referencing the Master display to now see the az_feature view mode?

Comment (trackleft) [inline modules/custom/az_news/config/install/views.view.az_news.yml:90]:
```suggestion
          view_mode: az_media_list
```
If not

**Review by cpoff — DISMISSED** [review]

**Comment by joeparsons** [comment]
> Fixed the media list issue, have a php error, and styling now isn't rendering as expected.

I think the PHP error was the result of someone trying to load the Probo site before it was fully built.  Just triggered a rebuild.

**Comment by danahertzberg** [comment]
Thanks! I'm making a couple more code changes and will be rebuilding again soon.

File: modules/custom/az_news/config/install/views.view.az_news.yml [hunk 2]
```diff
@@ -310,15 +378,230 @@ display:
...
+          id: area
+          table: views
+          field: area
+          relationship: none
+          group_type: group
+          admin_label: ''
+          empty: false
+          tokenize: false
+          content:
+            value: "<style>\r\n#views-exposed-form-az-news-az-grid-filter {\r\n    margin-bottom: 1.5em;\r\n}\r\n#views-exposed-form-az-news-az-grid-filter label {\r\n    font-weight: bold;\r\n}\r\n#views-exposed-form-az-news-az-grid-filter select {\r\n    color: #001c48;\r\n    font-weight: 500;\r\n    text-decoration: none;\r\n    letter-spacing: .04em;\r\n    white-space: normal;\r\n    border-width: 2px;\r\n    background-color: #dee2e6;\r\n    border-color: #dee2e6;\r\n    display: inline-block;\r\n    text-align: center;\r\n    vertical-align: middle;\r\n    -webkit-user-select: none;\r\n    -moz-user-select: none;\r\n    -ms-user-select: none;\r\n    user-select: none;\r\n    padding: 0.375rem 0.75rem;\r\n    font-size: 1rem;\r\n    line-height: 1.5;\r\n    border-radius: 0;\r\n    transition: color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;\r\n}\r\n</style>"
```
Comment (danahertzberg) [inline modules/custom/az_news/config/install/views.view.az_news.yml:452]:
```suggestion
            value: "<style>\r\n#views-exposed-form-az-news-az-grid-filter {\r\n    margin-bottom: 1.5em;\r\n}\r\n#views-exposed-form-az-news-az-grid-filter label {\r\n    font-weight: bold;\r\n    display: block;\r\n}\r\n#views-exposed-form-az-news-az-grid-filter select {\r\n    color: #001c48;\r\n    font-weight: 500;\r\n    text-decoration: none;\r\n    letter-spacing: .04em;\r\n    white-space: normal;\r\n    border-width: 2px;\r\n    background-color: #dee2e6;\r\n    border-color: #dee2e6;\r\n    display: inline-block;\r\n    text-align: center;\r\n    vertical-align: middle;\r\n    -webkit-user-select: none;\r\n    -moz-user-select: none;\r\n    -ms-user-select: none;\r\n    user-select: none;\r\n    padding: 0.375rem 0.75rem;\r\n    font-size: 1rem;\r\n    line-height: 1.5;\r\n    border-radius: 0;\r\n    transition: color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;\r\n}\r\n</style>"
```

File: modules/custom/az_news/config/install/views.view.az_news.yml (hunk 1 above)
Comment (trackleft) [reply modules/custom/az_news/config/install/views.view.az_news.yml:90]: I don't know if you have already done this, but we could also consider resaving all of the displays for the row settings marking them as "This display only" on save.

Comment (danahertzberg) [reply modules/custom/az_news/config/install/views.view.az_news.yml:90]: I did that for sure on paged row view and feature view! I believe the others were already like that, but I can double check.

Comment (trackleft) [reply modules/custom/az_news/config/install/views.view.az_news.yml:90]: image

**Comment by danahertzberg** [comment]
READY FOR FINAL REVIEW!

**Review by cpoff — DISMISSED** [review]

**Comment by smayden** [comment]
@danahertzberg 

I'm noticing that if I create a story with a caption/credit line that those appear in the feature view (but not the other views)

image

**Comment by danahertzberg** [comment]
> @danahertzberg
> 
> I'm noticing that if I create a story with a caption/credit line that those appear in the feature view (but not the other views)
> 

Good catch. I'll update.

**Review by trackleft — CHANGES_REQUESTED** [review]
Decision needs to be made about changing the master view display to use the az_feature display mode instead of the az_media_list display.

This will affect any existing site that is referencing the master display in a view paragraph.

I don't have an opinion about this other than we should acknowledge that it is an intended change.

**Comment by danahertzberg** [comment]
> Decision needs to be made about changing the master view display to use the az_feature display mode instead of the az_media_list display.
> 
> This will affect any existing site that is referencing the master display in a view paragraph.
> 
> I don't have an opinion about this, other than we should acknowledge that it is an intended change.

@trackleft Was this due to the reordering? I've always been perplexed by Drupal's "Master" view display that isn't configurable.. My opinion is to not change master. Sorry, I didn't realize reordering changes the master.

File: modules/custom/az_news/config/install/views.view.az_news.yml (hunk 1 above)
Comment (danahertzberg) [reply modules/custom/az_news/config/install/views.view.az_news.yml:90]: No :( I'll commit this. Thank you!

File: modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml [hunk 3]
```diff
@@ -198,7 +198,7 @@ content:
     weight: 6
     label: hidden
     settings:
-      view_mode: az_large
+      view_mode: az_card_image
```
Comment (trackleft) [inline modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml:201]: FYI, I'm fine with this, but be aware

**Before** 
image

**After**
image

**Review by trackleft — APPROVED** [review]

File: modules/custom/az_news/config/install/views.view.az_news.yml [hunk 4]
```diff
@@ -310,15 +378,230 @@ display:
...
+      block_category: 'Lists (Views)'
+      footer:
+        area:
+          id: area
+          table: views
+          field: area
+          relationship: none
+          group_type: group
+          admin_label: ''
+          empty: false
```
Comment (danahertzberg) [inline modules/custom/az_news/config/install/views.view.az_news.yml:449]:
```suggestion
          empty: true
```

File: modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml (hunk 3 above)
Comment (danahertzberg) [reply modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml:201]: Card image still looked good in my testing. It's also the only image style I'm aware of that doesn't show the image caption. Is there another style that doesn't show captions?

**Comment by smayden** [comment]
The marquee display use the "large" image, but does not display the caption (/news-teasers). I think this is because the card-img class is applied under manage display on the content type? I tried to recreate that in Probo but was not successful. 

I'd have to play a bit to see if there are any scaling issues with images. 

image

File: modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml (hunk 3 above)
Comment (danahertzberg) [reply modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml:201]: Ugh but card also has a crop, where large does not (original intention). Do we need a new image style?

Comment (danahertzberg) [reply modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml:201]: I'm okay keeping it as a card crop for now to not introduce a new image style.

**Comment by danahertzberg** [comment]
> The marquee display use the "large" image, but does not display the caption (/news-teasers). I think this is because the card-img class is applied under manage display on the content type? I tried to recreate that in Probo but was not successful.
> 
> I'd have to play a bit to see if there are any scaling issues with images.
> 

I tried this too with no luck. Follow up ticket?

**Comment by trackleft** [comment]
We don't have to use a rendered entity, I think since that field is an image only field, we should be fine just using an image style.

**Comment by trackleft** [comment]
image

**Comment by trackleft** [comment]
I changed it on probo, and it seems to look good to me.  https://7b5e2985-cedf-4a13-ab91-3a35916c835d--pr-1539.probo.build/admin/structure/types/manage/az_news/display/az_feature

```
<div class="field field--name-field-az-media-image field--type-entity-reference field--label-hidden field__item">  <img class="img-fluid image-style-az-large" src="/sites/default/files/styles/az_large/public/mans_best_friend.jpeg?itok=JkvEeU03" width="1140" height="763" alt="Cute dog" loading="lazy" typeof="foaf:Image">
</div>
```
https://7b5e2985-cedf-4a13-ab91-3a35916c835d--pr-1539.probo.build/dana

File: modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml [hunk 5]
```diff
@@ -198,7 +198,7 @@ content:
...
       link: false
```
Comment (danahertzberg) [inline modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml:202]:
```suggestion
      image_style: az_medium
      image_link: ''
```

**Comment by danahertzberg** [comment]
Thank you @trackleft! The "Thumbnail" format didn't register with me that I could change the image style without using the rendered image format. I changed it to use the medium image style instead of large to help performance and still achieve the original goal.

**Comment by danahertzberg** [comment]
- [ ] Update image style
- [ ] Follow up ticket for hiding filter if no terms
- [ ] Follow up ticket for reviewing current views/implementations

File: modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml [hunk 6]
```diff
@@ -198,8 +198,8 @@ content:
     weight: 6
     label: hidden
     settings:
-      view_mode: az_large
-      link: false
+      image_style: az_medium
+      image_link: ''
     third_party_settings: {  }
     type: entity_reference_entity_view
```
Comment (danahertzberg) [inline modules/custom/az_news/config/install/core.entity_view_display.node.az_news.az_feature.yml:204]:
```suggestion
    type: media_thumbnail
```

**Review by smayden — APPROVED** [review]

**Review by cpoff — APPROVED** [review]

**Review by camikazegreen — APPROVED** [review]

**Review by mmunro-ltrr — APPROVED** [review]

**Pull request merged**